CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+
+To create a widget on the dashboard, use the following code: 
+```
+widget = WidgetClass(...).create_widget()
+```
+
+Note: If you would like each plot to be linked to it's own individual widget, make sure you do not reuse widget objects for different plots - even if they have the same functionality. 
+
+
+## Creating a plot
+Each “plot” is an object of a specific class that inherits from a common, parent `Plot` class. Refer to the [`plot.py`](https://github.com/autogluon/autogluon-dashboard/blob/main/src/autogluon_dashboard/plotting/plot.py) file for more details regarding the attributes and methods contained within the parent class. The plots leverage the python library - `hvplot`. `hvplot` documentation can be found [here](https://hvplot.holoviz.org/user_guide/index.html). 
+
+To add a new plot, create a new file under `src/autogluon_dashboard/plotting` with appropriate naming conventions. Make sure that it inherits the parent `Plot` class. For the `plot` method, you can either leverage the inherited `_create_hvplot` method in the `Plot` class, the additional plot methods provided, or define a brand new plot function.
+
+If you would like to have an interactive plot with `panel` widgets included, you will need to pass in an interactive dataframe (which is an hvPlot representation) into the `hvplot` function. This can be done by calling the `interactive()` method on your `pandas` dataframe as: 
+```
+df = pd.read_csv('somefile.csv')
+idf = df.interactive()
+```
+You can then call the `.hvplot()` function as you would call `.plot()` on a pandas dataframe.
+
+
+## Adding a plot (w or w/o widgets) to the dashboard
+To add the plot to the dashboard, follow the steps used in `app.py`. 
+```
+# Global list of panel objects to display on the website
+panel_objs = []
+
+# Create an object of the plot class
+plot_obj = PlotClass(...)
+
+# Create an object of the widget class
+widget = WidgetClass(...).create_widget()
+
+# Add the panel object to global list
+create_panel_object(panel_objs, "some title", widgets=[widget], plots=[plot_obj])
+```
+
+Note: If you would like to pass in custom arguments to the `plot` method, you can create the plot in `app.py` itself; before passing it into `create_panel_object` as:
+```
+try: 
+    plot_obj = PlotClass(...).plot(arg=arg)
+except Exception:
+    plot_obj = None
+create_panel_object(panel_objs, "some title", plots=[plot_obj])
+```
+
+
+## Using the Dashboard wrapper locally
+Once you have installed the package from source as per the instructions in the [`README`](https://github.com/autogluon/autogluon-dashboard/blob/main/README.md), you should have access to the `agdash` command. 
+```
+agdash --help
+usage: agdash [-h] --per_dataset_csv  --agg_dataset_csv  [--hware_metrics_csv]  [--s3_bucket ] [--s3_prefix ] [--s3_region ]
```

Comment:
Do we want to explain where these csvs come from and what are they referring to? Maybe give an example on what they look like

Reply:
Sure, that's a good idea!